server/controllers/AuthController.js

```diff
@@ -48,32 +49,35 @@ module.exports = {
             return;
         }
         let authorized = false;
-        if (req.session.authType === SIGN_UP) {
-            const userSaved = await services.UsersService.signupUser({
-                accessToken: req.session.accessToken,
-                user: userToAuthorize
-            });
-            if (userSaved) {
-                authorized = true;
-                userToAuthorize = userSaved;
-            }
-        } else if (req.session.authType === LOGIN) {
-            const { name: fullname, username, email } = userToAuthorize;
+        if (
+            req.session.authType === SIGN_UP ||
+            req.session.authType === LOGIN
+        ) {
+            let user;
 
-            const dbUser = await services.UsersService.getUserAndUpdateRoles({
-                accessToken: req.session.accessToken,
-                user: {
+            if (req.session.authType === SIGN_UP) {
+                user = await UsersService.signupUser({
+                    accessToken: req.session.accessToken,
+                    user: userToAuthorize
+                });
+            } else {
+                const { name: fullname, username, email } = userToAuthorize;
+                user = {
                     fullname,
                     username,
                     email
-                }
-            });
-            if (dbUser) {
+                };
+            }
+
+            if (user) {
                 authorized = true;
-                userToAuthorize = dbUser;
+                userToAuthorize = await UsersService.getUserAndUpdateRoles({
```

Comment:
This function call fails if you try to log in, but the user doesn't exist... which is probably a problem with the function, but once we have a single "log in" button an a workflow that:
1. Checks to see if you have authenticated with github (takes you through the github authorization workflow if you haven't authorized github)
2. Checks to see if you are in the appropriate teams in github  
3. If you are in an appropriate team, checks to see if you are a user, makes a user for you if there isn't one
4. Updates your teams appropriately (in case they have changed)
5. If you are not in the appropriate teams, you should be redirected to the "how to get involved in this project" page (that doesn't currently exist)